### PR TITLE
refactor: Update to latest go-mod-secrets

### PR DESCRIPTION
**Pull request by lenny-intel**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-bootstrap/blob/master/.github/Contributing.md.

## What is the current behavior?
used previous verion of go-mod-secrets

## What is the new behavior?
Uses latest version of go-mod-secrets
 Adjusted for renamed client and addition of `Type` in Secret Store configuration

Dependent on PR https://github.com/edgexfoundry/go-mod-secrets/pull/91 to be merged

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

BREAKING CHANGE: All existing SecretStore configuration must add Type = 'vault'

## New Imports

- [ ] Yes
- [x] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Other information

### CODE REVIEW

File: bootstrap/handlers/secret.go [hunk 1]
```diff
@@ -103,6 +103,7 @@ func SecureProviderBootstrapHandler(
 // If a token file is present it will override the Authentication.AuthToken value.
 func getSecretConfig(secretStoreInfo config.SecretStoreInfo, tokenLoader authtokenloader.AuthTokenLoader) (types.SecretConfig, error) {
 	secretConfig := types.SecretConfig{
+		Type:                    secretStoreInfo.Type,
```
Comment (jim-wang-intel) [inline bootstrap/handlers/secret.go:106]: could you add a bit of description about what this `Type` mean?  developer might have hard time to figure out what type should be used...

Comment (lenny-intel) [reply bootstrap/handlers/secret.go]: Done

**Review by jim-wang-intel — APPROVED** [review]
LGTM

**Review by jim-wang-intel — APPROVED** [review]
LGTM

**Review by jpwhitemn — APPROVED** [review]

**Pull request merged**